Update doc.txt (#21136)
RIOT-OS/RIOT: pkg/paho-mqtt/doc.txt

/**
 @defgroup pkg_paho_mqtt   PAHO MQTT framework
 @ingroup  pkg
 @brief    The Eclipse Paho project provides open-source client implementations of MQTT for embedded systems

>  @see      https://github.com/eclipse/paho.mqtt.embedded-c

Reviewer: Looks good, so no need to change: https://ci.riot-os.org/results/715889b6444d498aa4654678cab2fb99/doc-preview/group__pkg__paho__mqtt.html

 
 The Eclipse Paho project provides open-source client

>  implementations of MQTT.

Reviewer 2: Isn't that redundant to the `@brief` text?

Author: It is duplicated. However, I only removed the asterisk here. The rest is original

Reviewer: I'd still remove the duplicated sentence here.

  # Supported MQTT Protocol Version

  The supported protocol version of the used paho.mqtt.embedded-c stack is
    - MQTT-Version = 3.1 (MQTTVersion struct member of MQTTPacket_connectData = 3)

> 	  - MQTT-Version = 3.1.1 (MQTTVersion struct member of MQTTPacket_connectData = 4)

Reviewer: So two different MQTT versions are supported? Then I would prefer plural in the sentence above.

Author: That's correct. But I don't know how to change this in this stage.

Reviewer: please see https://github.com/RIOT-OS/RIOT/pull/21176#issuecomment-2647513297

>   # Implementation specific information

Reviewer: ```suggestion
  # Implementation-specific information
```

>   - The Main-Loop is running in the Task mqtt_riot_run() with a default-cyclus of 30ms (MQTT_YIELD_POLLING_MS) 

Reviewer 2: ```suggestion
  - The main loop is running in a dedicated thread (` paho_mqtt_riot`) in the function `mqtt_riot_run()` with a default-cyclus of 30ms (MQTT_YIELD_POLLING_MS) 
```
I'm not sure if we have a well defined meaning for the term "Task" in RIOT.

    and an default duration of 10ms (PAHO_MQTT_YIELD_MS)

>   - During publish a topic with MQTTPublish() and QoS>0, the functions blocks until 
>     it receive des One Acknowledge (QoS=1) or two Acknowledges (Qos=2).

Reviewer 2: ```suggestion
  - When publishing a topic with `MQTTPublish()` and QoS>0, the functions blocks until 
    it one acknowledgement (QoS=1) or two acknowledgements (QoS=2) are received..
```
